docs: :memo: explanation of workflow to use on DST (#497)
steno-aarhus/osdc: vignettes/osdc.qmd

## Working with real register data {#sec-working-with-real-register-data}

In a real-world scenario, the register data is too large to read into
memory all at once. We recommend converting your register files into
[Parquet](https://parquet.apache.org/) format on disk, with each
register in its own folder (e.g., all `kontakter` files in one folder,
all `diagnoser` files in another, etc.).

If you're working on Statistics Denmark's (DST) server, be aware that
the pre-installed R packages on those servers can be quite old (for
example, at the time of writing, `library(duckplyr)` loads version 0.4
of the `duckplyr` package from 2024, whereas the latest version on CRAN
is \>1.1 from 2026). These old versions don't support the `dplyr`

> operations necessary for `osdc` to work.

Reviewer: ```suggestion
of the duckplyr package from 2024, whereas the latest version on CRAN
is \>1.1 from 2026). These old versions don't support the dplyr
operations necessary for osdc to work.
```

These packages all stylize their names without the backtick, as we do in our documentation. So just removing these backticks ☺️

> Installing `osdc` with `install.packages("osdc")` should force the

Reviewer: ```suggestion
Installing osdc with `install.packages("osdc")` should force the
```

necessary updates of package dependencies. Otherwise, the latest version
of any given package can be installed from DST's local CRAN mirror by
using `install.packages()`. The downside is that you have to repeat the
package installations frequently (whenever you log on to a new virtual
machine, or after the servers weekly reset).

> After making sure that you have the newest version of `osdc` (which

Reviewer: ```suggestion
After making sure that you have the newest version of osdc (which
```

should install the updated dependencies) you can then load each register
directly from its Parquet folder and convert it to a DuckDB table,
without reading everything into memory. If you're working with very
large data sets, it's good practice to filter this data a bit on the
Arrow side before converting to DuckDB to reduce computation and memory

> use. This is convenient, as most of `dplyr`'s filtering and selection

Reviewer: ```suggestion
use. This is convenient, as most of dplyr's filtering and selection
```

> operations are supported by the `arrow` package. If you need to rename

Reviewer: ```suggestion
operations are supported by the arrow package. If you need to rename
```

variables or convert their types to match the inputs expected by `osdc`,

> this is also an opportunity to do so.:

Reviewer: ```suggestion
this is also an opportunity to do so:
```

``` r
diagnoser <- "path/to/diagnoser_parquet_folder" |>
  arrow::open_dataset(unify_schemas = TRUE) |>
  arrow::to_duckdb() |>
  # Insert dplyr filtering steps here, as appropriate, e.g.:
  # dplyr::select(...) |> dplyr::filter(...) |> 

>   arrow::to_duckdb() |> dplyr::compute()

Reviewer: ```suggestion
  arrow::to_duckdb() |> 
  dplyr::compute()
```

## Working with real register data {#sec-working-with-real-register-data}

In a real-world scenario, the register data is too large to read into
memory all at once. We recommend converting your register files into
[Parquet](https://parquet.apache.org/) format on disk, with each
register in its own folder (e.g., all `kontakter` files in one folder,
all `diagnoser` files in another, etc.).

> 

Reviewer 2: #ad

```suggestion
all `diagnoser` files in another, etc.).

::: callout-tip
To convert SAS (`.sas7bdat`) files to Parquet, you can use the [`fastreg`](https://dp-next.github.io/fastreg/) package.
:::

```

If you're working on Statistics Denmark's (DST) server, be aware that

> the pre-installed R packages on those servers can be quite old (for

Reviewer 2: ```suggestion
the pre-installed R packages can be quite old (for
```

example, at the time of writing, `library(duckplyr)` loads version 0.4
of the duckplyr package from 2024, whereas the latest version on CRAN
is \>1.1 from 2026). These old versions don't support the dplyr
operations necessary for osdc to work.

Installing osdc with `install.packages("osdc")` should force the
necessary updates of package dependencies. Otherwise, the latest version
of any given package can be installed from DST's local CRAN mirror by

> using `install.packages()`. The downside is that you have to repeat the

Reviewer 2: ```suggestion
using `install.packages()`. The downside of this approach is that you have to repeat the
```

package installations frequently (whenever you log on to a new virtual
machine, or after the servers weekly reset).

After making sure that you have the newest version of osdc (which

> should install the updated dependencies) you can then load each register

Reviewer 2: ```suggestion
should install the updated dependencies), you can load each register
```

directly from its Parquet folder and convert it to a DuckDB table,

> without reading everything into memory. If you're working with very

Reviewer 2: ```suggestion
without reading it into memory. If you're working with very
```

large data sets, it's good practice to filter this data a bit on the

> Arrow side before converting to DuckDB to reduce computation and memory

Reviewer 2: ```suggestion
large data sets, it's good practice to filter this data on the
Arrow side before converting to DuckDB to reduce computation and memory
```

use. This is convenient, as most of dplyr's filtering and selection

> operations are supported by the arrow package. If you need to rename

Reviewer 2: ```suggestion
use. Conveniently, most of dplyr's filtering and selection
operations are supported by the arrow package. If you need to rename
```

> variables or convert their types to match the inputs expected by `osdc`,

Reviewer 2: ```suggestion
variables or convert their types to match the inputs expected by osdc,
```

> this is also an opportunity to do so:

Reviewer 2: ```suggestion
you can do that too:
```

``` r
diagnoser <- "path/to/diagnoser_parquet_folder" |>
  arrow::open_dataset(unify_schemas = TRUE) |>
  arrow::to_duckdb() |>
  # Insert dplyr filtering steps here, as appropriate, e.g.:

>   # dplyr::select(...) |> dplyr::filter(...) |> 

Reviewer 2: ```suggestion
  # dplyr::select(...) |> 
  # dplyr::filter(...) |> 
```

format.

> :::

Author: ```suggestion
```
